driver/others/blas_server.c

```diff
@@ -566,12 +568,9 @@ int blas_thread_init(void){
 
     for(i = 0; i < blas_num_threads - 1; i++){
 
-      thread_status[i].queue  = (blas_queue_t *)NULL;
+      atomic_store_queue(&thread_status[i].queue, (blas_queue_t *)0);
       thread_status[i].status = THREAD_STATUS_WAKEUP;
 
-      pthread_mutex_init(&thread_status[i].lock, NULL);
-      pthread_cond_init (&thread_status[i].wakeup, NULL);
```

Comment:
@AGSaidi, `thread_status.lock` and `thread_status.wakeup` are now being used uninitialized in the code.  This may explain the odd garbage values seen and the cause of the hang.

Reply:
Good catch, don't know how often I have looked over that diff without spotting it. Sanity is indeed restored by restoring those two lines.

Reply:
Applied the changes to confirm this fixes the CI failures as well...

Reply:
Thanks @geoffreyblake. I too don't understand how I missed them